tests/testutil.py

```diff
@@ -31,15 +32,26 @@ def __exit__(self, exc_Type, exc_value, traceback):
     if self.cwd is not None:
       os.chdir(self.cwd)
 
-def exec_argv(argv):
+def exec_argv(argv, logger = None):
   if argv[0].endswith('.js'):
     argv = ['node'] + argv
 
   p = subprocess.Popen(argv, stdout = subprocess.PIPE, stderr = subprocess.PIPE)
```

Comment:
Mentioned in IRC; but I'll throw this here. 

Popen has a timeout parameter that throws an Exception when that timeout is reached. Perhaps that's better?

Reply:
Agree it would be better, but I think that's only available in a totally separate, unrelated language called Python 3.